File: frontend/src/backend/seasonService.js
import {config, appFetch} from './appFetch';

export const findAllSeasons = (onSuccess) =>
    appFetch('/seasons', config('GET'), onSuccess);

export const findSeasonById = (id, onSuccess) =>
    appFetch(`/seasons/${id}`, config('GET'), onSuccess);

export const findSeasonsBetweenTwoDates = (startDate, endDate, onSuccess, onErrors) =>
    appFetch(`/seasons/dates?startDate=${startDate}&endDate=${endDate}`, config('GET'), onSuccess, onErrors);

export const addSeason = (startDate, endDate, calendario, onSuccess, onErrors) =>
    appFetch(`/seasons?startDate=${startDate}&endDate=${endDate}&calendario=${calendario}`, config('POST'), onSuccess, onErrors);

export const updateSeason = (id, startDate, endDate, calendario, onSuccess, onErrors) =>
    appFetch(`/seasons/${id}?startDate=${startDate}&endDate=${endDate}&calendario=${calendario}`, config('PUT'), onSuccess, onErrors);

export const removeSeason = (id, onSuccess, onErrors) =>
    appFetch(`/seasons/${id}`, config('DELETE'), onSuccess, onErrors);

export const findSeasonsToTeam = (teamId, onSuccess, onErrors) => 
    appFetch(`/seasons/${teamId}/team`, config('GET'), onSuccess, onErrors);

File: backend/src/main/java/es/udc/paproject/backend/model/services/SeasonService.java
package es.udc.paproject.backend.model.services;

import java.time.LocalDateTime;
import java.util.List;

import es.udc.paproject.backend.model.entities.Season;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.StartDateAfterEndDateException;

public interface SeasonService {

    Season addSeason(Long userId, LocalDateTime startDate, LocalDateTime endDate, String calendario) throws InstanceNotFoundException, StartDateAfterEndDateException;

    Season findSeasonById(Long userId, Long seasonId) throws InstanceNotFoundException;

    List<Season> findSeasonsBetweenTwoDates(Long userId, LocalDateTime startDate, LocalDateTime endDate) throws InstanceNotFoundException, StartDateAfterEndDateException;

    List<Season> findAllSeasons(Long userId) throws InstanceNotFoundException;
    
    List<Season> findSeasonsToTeam(Long userId, Long teamId) throws InstanceNotFoundException;

    void removeSeason(Long userId, Long seasonId) throws InstanceNotFoundException;
    
    void clearSeasonTeamTable(Long userId) throws InstanceNotFoundException;

    Season updateSeason(Long userId, Long seasonId, LocalDateTime startDate, LocalDateTime endDate, String calendario) throws InstanceNotFoundException;
}